backend/api/src/main/kotlin/de/keepmealive3d/event/LiveDataEventHandler.kt

```diff
@@ -1,7 +1,34 @@
 package de.keepmealive3d.event
 
-class LiveDataEventHandler: EventHandler {
+import de.keepmealive3d.adapters.influx.KmaInfluxDatabase
+import kotlinx.coroutines.channels.consumeEach
+import kotlinx.coroutines.flow.consumeAsFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.koin.core.component.KoinComponent
+
+class LiveDataEventHandler : EventHandler, KoinComponent {
+    private val database = KmaInfluxDatabase()
+
+
     fun receive(topic: String, payload: String) {
+
+        runBlocking {
+            launch {
+                try {
+                    // TODO: write actual data
+                    database.write("mem", "test", 42.0)
+
+                    database.readAllMeasurements("mem").consumeAsFlow().collect { println("${it.value}") }
+
+                } catch (exception: Exception) {
+                    println(exception)
+                }
+
+            }
+        }
+
+
         println("Received on topic: $topic -> $payload")
     }
```

Comment:
The launch {} has no effect if wrapped in a runBlocking. I would do something like this:
```suggestion
    suspend fun receive(topic: String, payload: String) = withContext(Dispatchers.IO) {
                try {
            // TODO: write actual data
            database.write("mem", "test", 42.0)

            database.readAllMeasurements("mem").consumeAsFlow().collect { println("${it.value}") }

        } catch (exception: Exception) {
            println(exception)
        }
        println("Received on topic: $topic -> $payload")
    }
``` 
Instead of `withContext(Dispatchers.IO)` you could also just use `coroutineScope { }` but with setting the context explicitly, kotlin can do some optimizations.

Reply:
you also have to add the suspend to the rcv function in the Plugin class `rcv: suspend (topic: String, value: String) -> Unit,`